tests/getmempolicy/getmempolicy.c

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <assert.h>
+#include <errno.h>
+#include <limits.h>
+#include <sched.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syscall.h>
+#include <unistd.h>
+
+#define MPOL_DEFAULT 0
+#define MPOL_PREFERRED 1
+#define MPOL_BIND 2
+#define MPOL_INTERLEAVE 3
+#define MPOL_LOCAL 4
+#define MPOL_MAX 5
+
+#define MPOL_F_NODE (1 << 0)
+#define MPOL_F_ADDR (1 << 1)
+#define MPOL_F_MEMS_ALLOWED (1 << 2)
+
+#define MPOL_MF_STRICT (1 << 0)
+#define MPOL_MF_MOVE (1 << 1)
+#define MPOL_MF_MOVE_ALL (1 << 2)
+
+long get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    return syscall(SYS_get_mempolicy, mode, nodemask, maxnode, addr, flags);
+}
+
+void print(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    printf(
+        "\nmode = %d\tnodemask = %ld\tmaxnode = %ld\taddr = %p\tflags = %lu",
+        *mode,
+        *nodemask,
+        maxnode,
+        addr,
+        flags);
+}
+
+int test1()
+{
+    int mode = 0xffffffff;
+    unsigned long nodemask = 0xffffffffffffffff;
+    unsigned long maxnode = 64;
+    void* addr = NULL;
+    unsigned long flags = MPOL_DEFAULT;
+
+    printf("1. MPOL_DEFAULT, null addr");
+    print(&mode, &nodemask, maxnode, addr, flags);
+    long ret = get_mempolicy(&mode, &nodemask, maxnode, addr, flags);
+    print(&mode, &nodemask, maxnode, addr, flags);
+    printf("\nRet = %ld", ret);
+    printf("\nerrno = %d", errno);
+
+    assert(ret == 0);
+    assert(mode == 0);
+    assert(nodemask == 0);
+    assert(maxnode == 64);
```

Comment:
The check has no effect, as maxnode is an input set to 64. Same comments on other assert code on input variables.